src/renderers/shared/fiber/ReactFiberClassComponent.js

```diff
@@ -338,14 +361,16 @@ module.exports = function(
     // componentWillMount and before this componentWillMount? Probably
     // unsupported anyway.
 
-    if (!checkShouldComponentUpdate(
-      workInProgress,
-      workInProgress.memoizedProps,
-      newProps,
-      workInProgress.memoizedState,
-      newState,
-      newContext
-    )) {
+    if (
+      !checkShouldComponentUpdate(
+        workInProgress,
+        workInProgress.memoizedProps,
+        newProps,
+        workInProgress.memoizedState,
+        newState,
+        newContext,
+      )
```

Comment:
It's consistent with breaking a function call; if the args are too long it will always put arguments on their own lines. We could see if there is a way to keep it on the same line if there is only one argument, which would also always apply to if/while/etc statements.

We could at least implement this heuristic: if there is only one element inside parens, and it's a unary operator or a function call (maybe other things), print it on the same line. If we do it more generally we'll see things like:

```js
if(thisCondition &&
  thatCondition &&
  anotherCondition) {
  const x = 5;
}
```

Which is really bad; it's a lot more clear to move the whole expression to the same line (same reason a lot of people put the end bracket in JSX on its own line).

But there's no reason we couldn't optimize for a few patterns. Is it important enough to do so?